### PR TITLE
Update smoke effect

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Smoke filed apply different intensities of smoke effect on player. However, the effect contain no intensities, so it is not used. While here, polish the effect and make it better overall
#### Describe the solution
Now there is two effects, first respond for eye irritating, and can be countered with eye protection, and second for breathing issues, that can be countered by mouth protection
Each effect apply it's own set of effects, that progress the worse smoke field you happen to enter
Effect uses #65549 to write different messages for different intensities, and #66683 for multiple `immunity_data`s

### CODE REVIEW

**Comment by Qrox** [comment]
While you are at it could you also address https://github.com/CleverRaven/Cataclysm-DDA/pull/66291#issuecomment-1596409761? I.e. the smoked effect should also mention eye irritation in addition to lung irritation.

**Comment by GuardianDll** [comment]
That's actually the reason i didn't make it sooner: fd_smoke specify only `mouth` bodypart in immunity data
https://github.com/CleverRaven/Cataclysm-DDA/blob/a71a91179f84802a3c081ffd05339936355abd69/data/json/field_type.json#L505
If there was a way to split effect into moutn and eye parts, where eye effect checks against env stat of eyes, and mouth effect checks against env stat of mouth, it would work much better. But, from what i know, it's not possible, and i don't really know should i make this changes or wait when such thing would be possible

File: data/json/effects.json [hunk 1]
```diff
@@ -1779,15 +1779,70 @@
...
+      "Your eyes tear and stings heavily.",
+      "Your eyes burn out of smoke, and you hardly see anything."
+    ],
+    "apply_message": [
+      [ "Your eyes mildly irritate.", "bad" ],
+      [ "Your eyes watering out of smoke.", "bad" ],
+      [ "Your eyes tear and stings heavily.", "bad" ],
+      [ "Your eyes burn out of smoke, and you hardly see anything.", "bad" ]
+    ],
+    "rating": "bad",
```
Comment (GuardianDll) [inline data/json/effects.json:1801]:
```suggestion
```

File: data/json/effects.json [hunk 2]
```diff
@@ -1779,15 +1779,70 @@
...
+      "Dense cloud of smoke enter your lungs.",
+      "You inhale an overwhelming amount of smoke."
+    ],
+    "apply_message": [
+      [ "You draw in a faint whiff of smoke.", "bad" ],
+      [ "Noticeable amount of smoke fills your lungs.", "bad" ],
+      [ "Dense cloud of smoke enter your lungs.", "bad" ],
+      [ "You inhale an overwhelming amount of smoke.", "bad" ]
+    ],
     "rating": "bad",
```
Comment (GuardianDll) [inline data/json/effects.json:1831]:
```suggestion
```

File: data/json/field_type.json [hunk 3]
```diff
@@ -454,11 +454,21 @@
         "translucency": 1,
         "effects": [
           {
-            "effect_id": "smoke",
+            "effect_id": "smoke_eyes",
+            "body_part": "eyes",
+            "intensity": 1,
+            "min_duration": "2 seconds",
+            "max_duration": "2 seconds",
+            "immunity_data": { "body_part_env_resistance": [ [ "eyes", 3 ] ] },
```
Comment (GuardianDll) [inline data/json/field_type.json:462]:
```suggestion
            "immunity_data": { "body_part_env_resistance": [ [ "sensor", 3 ] ] },
```

File: data/json/field_type.json [hunk 4]
```diff
@@ -470,11 +480,21 @@
         "translucency": 10,
         "effects": [
           {
-            "effect_id": "smoke",
+            "effect_id": "smoke_eyes",
+            "body_part": "eyes",
+            "intensity": 2,
+            "min_duration": "7 seconds",
+            "max_duration": "7 seconds",
+            "immunity_data": { "body_part_env_resistance": [ [ "eyes", 5 ] ] },
```
Comment (GuardianDll) [inline data/json/field_type.json:488]:
```suggestion
            "immunity_data": { "body_part_env_resistance": [ [ "sensor", 5 ] ] },
```

File: data/json/field_type.json [hunk 5]
```diff
@@ -486,11 +506,21 @@
         "translucency": 0,
         "effects": [
           {
-            "effect_id": "smoke",
+            "effect_id": "smoke_eyes",
+            "body_part": "eyes",
+            "intensity": 4,
+            "min_duration": "15 seconds",
+            "max_duration": "15 seconds",
+            "immunity_data": { "body_part_env_resistance": [ [ "eyes", 7 ] ] },
```
Comment (GuardianDll) [inline data/json/field_type.json:514]:
```suggestion
            "immunity_data": { "body_part_env_resistance": [ [ "sensor", 7 ] ] },
```

**Comment by Kamayana** [comment]
I think this PR removing the `"smoke"` effect is causing crashes (#66980 #67005), as it seems like the game is looking for an effect that doesn't exist, and it doesn't look like there's a migration of `"smoke"` to the new `"smoke_lung"` (though I'm not sure if migrating smoke effects is actually possible).

**Comment by SurFlurer** [comment]
A no-longer-exist effect will cause crash when the game wanna translate its descriptions / messages. Preventing this kind of crash will solve #66255.

**Pull request merged**